react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      index: 0
+    };
+    this.arrowClicks = this.arrowClicks.bind(this);
+    this.renderProgressDots = this.renderProgressDots.bind(this);
+    this.clickDots = this.clickDots.bind(this);
+    this.updateCarousel = this.updateCarousel.bind(this);
+  }
+
+  componentDidMount() {
+    this.updateCarousel();
+  }
+
+  updateCarousel() {
+    const intervalId = setInterval(() => {
+      if (this.state.index === this.props.images.length - 1) {
+        this.setState({
+          index: 0
+        });
+      } else {
+        this.setState({
+          index: this.state.index + 1
+        });
+      }
+    }, 3000);
+    return intervalId;
+  }
+
+  arrowClicks(event) {
+    clearInterval(this.updateCarousel());
+    const index = this.state.index;
+    if (event.target.id === 'right') {
+      if (index === this.props.images.length - 1) {
+        this.setState({
+          index: 0
+        });
+      } else {
+        this.setState({
+          index: index + 1
+        });
+      }
+    }
+
+    if (event.target.id === 'left') {
+      if (index === 0) {
+        this.setState({
+          index: this.props.images.length - 1
+        });
+      } else {
+        this.setState({
+          index: index - 1
+        });
+      }
+    }
```

Comment:
Then at the end of this method, call `this.updateInterval()` to establish a new interval id on `this` to be cleared later.